x-pack/plugins/maps/public/classes/sources/mvt_single_layer_vector_source/mvt_single_layer_source_settings.tsx

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
+import React, { Fragment, Component, ChangeEvent } from 'react';
+import { EuiFieldText, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import _ from 'lodash';
+import { MAX_ZOOM, MIN_ZOOM } from '../../../../common/constants';
+import { ValidatedDualRange, Value } from '../../../../../../../src/plugins/kibana_react/public';
+import { MVTFieldConfigEditor } from './mvt_field_config_editor';
+import { MVTFieldDescriptor } from '../../../../common/descriptor_types';
+
+export type MVTSettings = {
+  layerName: string;
+  fields: MVTFieldDescriptor[];
+  minSourceZoom: number;
+  maxSourceZoom: number;
+};
+
+export interface State {
+  currentLayerName: string;
+  currentMinSourceZoom: number;
+  currentMaxSourceZoom: number;
+  currentFields: MVTFieldDescriptor[];
+  previousLayerName: string;
+  previousMinSourceZoom: number;
+  previousMaxSourceZoom: number;
+  previousFields: MVTFieldDescriptor[];
+}
+
+export interface Props {
+  handleChange: (args: MVTSettings) => void;
+  layerName: string;
+  fields: MVTFieldDescriptor[];
+  minSourceZoom: number;
+  maxSourceZoom: number;
+  includeFields: boolean;
```

Comment:
`includeFields` is a little vague. What are fields getting included in?

Variables that toggle display are typically prefixed with `show`. How about renaming to `showFields`